packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,401 @@
+
+/// A delegate which the [RasterWidget] can use to fallback to regular rendering
+/// if a platform view is present in the layer tree.
+///
+/// Consumers of [RasterWidget] should almost never use this delegate. For the most part,
+/// the raster widget only functions as a performance improvement. If a platform view is
+/// present, the performance improving qualities aren't possible and using this API is
+/// pointless.
+///
+/// Instead, this interface is useful if a generic/reusable widget is being created which
+/// may include a platform view and it needs to handle this transparently. For example, the
+/// framework uses this for the zoom page transition so that navigating to a page shows the same
+/// animation whether or not there is a platform view.
+abstract class RasterWidgetFallbackDelegate {
+  /// const constructor so that subclasses can be const.
+  const RasterWidgetFallbackDelegate();
+
+  /// Paint the child via [painter], applying any effects that would have been painted
+  /// with the [RasterWidgetDelegate].
+  ///
+  /// The [offset] and [size] are the location and dimensions of the render object.
+  void paintFallback(PaintingContext context, Offset offset, Size size, PaintingContextCallback painter);
+}
+
+/// A widget that replaces its child with a rasterized version of the child.
+///
+/// By default, the child is drawn as is. The default [delegate] simply scales
+/// down the image by the current device pixel ratio and paints it into the
+/// canvas. How this image is drawn can be customized by providing a new
+/// subclass of [RasterWidgetDelegate] to the [delegate] argument.
+///
+/// Caveats:
+///
+/// The contents of platform views cannot be captured by a raster
+/// widget. If a platform view is encountered, then nothing will be rendered unless
+/// [ignorePlatformViews] is `true` or a [fallback] delegate is provided.
+///
+/// This widget is not supported on the HTML backend of Flutter for the web.
+class RasterWidget extends SingleChildRenderObjectWidget {
+  /// Create a new [RasterWidget].
+  ///
+  /// The [controller] and [child] arguments are required.
+  const RasterWidget({
+    super.key,
+    this.delegate = const _RasterDefaultDelegate(),
+    this.fallback,
+    this.ignorePlatformViews = false,
+    required this.controller,
+    required super.child
+  });
+
+  /// A delegate that allows customization of how the image is painted.
+  ///
+  /// If not provided, defaults to a delegate which paints the child as is.
+  final RasterWidgetDelegate delegate;
+
+  /// The controller that determines when to display the children as an image.
+  final RasterWidgetController controller;
+
+  /// A fallback delegate which is used if the child layers contains a platform view.
+  final RasterWidgetFallbackDelegate? fallback;
+
+  /// Whether the raster widget will ignore platform views in child layers.
+  ///
+  /// Defaults to `false`, which causes an error to be thrown if a platform view
+  /// is encountered.
```

Comment:
... unless a fallback has been specified.

It is rather complicate to wrap one's head around how `ignorePlatformViews` and `fallback` interact. Do we really need this as separate settings? Could we do with just the optional `fallback` and if you don't provide one, we just error to simplify things? Why dod we need the option to ignore platform views?

Reply:
I added some more context to the documentation here. The idea is, suppose that you are using this widget and the child has a platform view. But the platform view isn't really important, and you don't care if its included in the scene. Without this flag, there is no way to force rasterization of the scene. You can provide a fallback but that doesn't allow you to opt into rasterization

Reply:
Can you add exactly that explanation to the doc?